### PR TITLE
don't sync the whole product if the repository could not be found

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: dab1c937eb379aa7e77fd24b00ee34f6ec84b9ea

### CODE REVIEW

**Review by pmoravec — APPROVED** [review]
Works great in all three scenarios:

1) just product name supplied (whole product synced)
2) product name and repo name correctly supplied (exactly the repo synced)
3) valid product name and invalid repo name supplied (originally, whole product synced, newly `"Found no results while searching for repositories with name=\"invalid repo name\""` reported as a failure)

File: tests/test_playbooks/fixtures/repository_ostree-1.yml [hunk 1]
```diff
@@ -253,13 +248,20 @@ interactions:
       User-Agent:
       - apypie (https://github.com/Apipie/apypie)
     method: POST
-    uri: https://foreman.example.org/katello/api/products/4/sync
+    uri: https://foreman.example.org/katello/api/repositories/229/sync
```
Comment (evgeni) [inline tests/test_playbooks/fixtures/repository_ostree-1.yml:251]: This is exactly the bug that this PR is fixing ;)

File: plugins/modules/repository_sync.py [hunk 2]
```diff
@@ -66,9 +66,7 @@ def main():
     module = KatelloAnsibleModule(
         foreman_spec=dict(
             product=dict(type='entity', scope=['organization'], required=True),
-            repository=dict(type='entity', scope=['product'], failsafe=True),
-            # This should be scoped more explicit for better serch performance, but needs rerecording
-            # repository=dict(type='entity', scope=['organization', 'product'], failsafe=True),
+            repository=dict(type='entity', scope=['product']),
```
Comment (mdellweg) [inline plugins/modules/repository_sync.py:69]: I get, the `failsafe` was the issue. What about the "better scoping" was that real?

Comment (evgeni) [reply plugins/modules/repository_sync.py:69]: No, the commit that dropped the comment says:
> the product is already scoped by org, so there is no benefit of scoping
> the repository by both product *and* org.

File: tests/test_playbooks/repository_ostree.yml [hunk 3]
```diff
@@ -40,7 +40,7 @@
     - name: sync repository
       include_tasks: tasks/katello_sync.yml
       vars:
-        repository: "Test Ostree Repository"
+        repository: "Test OStree Repository"
```
Comment (mdellweg) [inline tests/test_playbooks/repository_ostree.yml:43]: So this typo triggered the bug and we never noticed?
Should we keep a version of this expecting to properly fail now?

**Review by mdellweg — APPROVED** [review]
Just a suggestion for a test. Approving anyway.

File: tests/test_playbooks/repository_ostree.yml (hunk 3 above)
Comment (evgeni) [reply tests/test_playbooks/repository_ostree.yml:43]: Correct, the typo triggered the wrong codepath being executed.

I'll merge as-is and add a test in a separate PR.

Comment (evgeni) [reply tests/test_playbooks/repository_ostree.yml:43]: https://github.com/theforeman/foreman-ansible-modules/pull/1532

**Pull request merged**